src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,160 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        Node<T> prev;
+        T value;
+        Node<T> next;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public MyLinkedList() {
+        tail = new Node<>(null, null, null);
+        head = tail;
+        size = 0;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (size == 0) {
+            tail.value = value;
+            size++;
+            return true;
+        }
+        Node<T> newNode = new Node<>(tail, value, null);
+        tail.next = newNode;
+        tail = newNode;
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        indexCheck(index);
+        if (index == 0) {
+            head = new Node<>(null, value, head);
+            size++;
+            return;
+        }
+        Node<T> currentNode = findNodeByIndex(index);
+        Node<T> newNode = new Node<>(currentNode.prev, value, currentNode);
+        newNode.prev.next = newNode;
+        currentNode.prev = newNode;
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T t : list) {
+            add(t);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexCheck(index);
+        Node<T> node = findNodeByIndex(index);
+        if (node == head) {
+            return head.value;
+        }
+        return node.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        indexCheck(index);
+        Node<T> node = findNodeByIndex(index);
+        T temp = node.value;
+        node.value = value;
+        return temp;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexCheck(index);
+        Node<T> node = findNodeByIndex(index);
+        T temp = node.value;
+        if (size == 1) {
+            head.value = null;
+        } else {
+            if (node == head) {
+                head.next.prev = null;
+                head = head.next;
+            } else if (node == tail) {
+                tail.prev.next = null;
+                tail = tail.prev;
+            } else {
+                node.prev.next = node.next;
+                node.next.prev = node.prev;
+            }
+        }
+        size--;
+        return temp;
```

Comment:
let's move this logic to the separate private method unlink/delete() and use it in the methods remove by index and remove by object